examples/MMF-activity-recognition/README.md

```diff
@@ -0,0 +1,130 @@
+
+### Serving Activity Recognition MMF Model with Torchserve
+
+This section, we have the trained MMF model for activity recognition, it can be served in production with [Torchserve](https://github.com/pytorch/serve). 
+
+To serve a model using Torchserve, we need to bundle the model artifacts and a handler into a mar file which is an archive format that torchserve uses to serve our model, model_archiver package does this step. The mar file will get extracted in a temp directory and the Path will be added to the PYTHONPATH.
+
+#### Requirements
+
+Install [Torchserve](https://github.com/pytorch/serve)
+
+
+Install [MMF](https://mmf.sh/docs#install-from-source-recommended) from source or install latest MMF version from GitHub.
+
+`pip install git+https://github.com/facebookresearch/mmf.git`
+
+If you installed using pip, then you need install Pyav :
+
+`pip install av`
+
+MMF currenly is using Transformers 3.4.0, in case you have other version installed in your enviroment, this would be the best instead of installing it directly, add the MMF Github and 'av', in the requirements.txt and pass it to the model archiver using -r flag. You can read more about serving models with thrid party dependencies [here](https://github.com/pytorch/serve/blob/master/docs/use_cases.md#serve-custom-models-with-third-party-dependency).
+
+#### Getting started on Serving
+
+To make the mar file, we will use model_archiver as follows.
+
+After training the MMF model, the final checkpoints are saved in the mmf/save/ directory, where we need to use it as the serialized file for model-archiver. The [Charades action labels](https://mmfartifacts.s3-us-west-2.amazonaws.com/charades_action_lables.csv) along with the config file that covers all the settings of this experiment (can be found in mmf/save/config.yaml in case of training) are passed as extra files, they are getting used in the handler. Please make sure to change the label and config files accordingly in the handler (Lines #53, #68) if you are passing different ones.
+
+**You can simply download the mar file as follows:**
+
+`wget https://mmfartifacts.s3-us-west-2.amazonaws.com/MMF_activity_recognition.mar`
+
+ If mar file is downloaded then skip this and move to the next step. The other option is to download a pre-trained model, along with labels and config for this example and package them to a mar file. 
+
+```
+wget https://mmfartifacts.s3-us-west-2.amazonaws.com/mmf_transformer_Charades_final.pth
+wget https://mmfartifacts.s3-us-west-2.amazonaws.com/charades_action_lables.csv
+wget https://mmfartifacts.s3-us-west-2.amazonaws.com/config.yaml
+```
+
+```
+torch-model-archiver --model-name MMF_activity_recognition --version 1.0 --serialized-file $(path to checkpoints) --handler handler.py --extra-files "charades_action_lables.csv,$(path to the aggregated config file,saved in mmf/save/config.yaml)"
```

Comment:
Is there a pre-trained model I can use? If not maybe put these steps after the model_training section so it's clearer which order they need to be run in

Reply:
Thanks @msaroufim, I added the new mar file after changes, tested it works on my end. 
Then there is a pre-trained model that weights get downloaded here and then I modified the archiver to use the checkpoints and config.yaml that we download. 

`wget https://mmfartifacts.s3-us-west-2.amazonaws.com/mmf_transformer_Charades_final.pth`

Then at the last part we will have the training section. Adding the logs [here](https://gist.github.com/HamidShojanazeri/88f4ccf2bb3316b525e8ca2467ecfe2a).